features/dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -43,9 +49,10 @@ void write_crash_report(int signum,
     static const std::string crash_log_filename = "crash_log";
 
     // sync everything
-    pthread_mutex_lock(&handler_mutex);
     if (tracking_consent != tracking_consent_granted) {
```

Comment:
Is there any reason we can't move this check up before the mutex lock? I wouldn't think we'd care about a race condition here.